### PR TITLE
Jinja2 support in labels; conditional export at metric level; fixes for z16

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.

### CODE REVIEW

File: docs/appendix.rst [hunk 1]
```diff
@@ -32,6 +32,16 @@ Glossary
    Z HMC
       Hardware Management Console for IBM Z
 
+   Jinja2 expression
+     A Jinja2 expression (without curly braces).
+     See https://jinja.palletsprojects.com/en/3.1.x/templates/#expressions
+     for a description.
+
+     Note that when putting Jinja2 expression into YAML files, it is recommended
```
Comment (simon-spinner) [inline docs/appendix.rst:40]: grammar: ">a< Jinja2 expression" or "Jinja2 expression>s<"

File: docs/usage.rst [hunk 2]
```diff
@@ -588,6 +588,120 @@ Legend:
...
+
+* ``{label-value}`` - A :term:`Jinja2 expression` that is evaluated and used as
+  the value of the label.
+
+  The following variables are available for use in that Jinja2 expression:
+
+  - ``resource_obj`` - The resource to which the metric applies, as a
+    :class:`zhmcclient.BaseResource` object. This can be used for example to
+    get the name of the resource or its parent or grand-parent resources,
+    or to get other propertes of the resource.
```
Comment (simon-spinner) [inline docs/usage.rst:656]: propertes -> properties

File: docs/usage.rst [hunk 3]
```diff
@@ -588,6 +588,120 @@ Legend:
...
+
+  - ``metric_values`` - Dictionary of all metric values in the same metric group,
+    for the resource to which the metric applies. Key is the HMC metric name.
+    This variable is only present for HMC metrics based metrics, but not for
```
Comment (simon-spinner) [inline docs/usage.rst:660]: HMC-based metrics?

**Review by simon-spinner — APPROVED** [review]

File: docs/appendix.rst (hunk 1 above)
Comment (andy-maier) [reply docs/appendix.rst:40]: DONE

File: docs/usage.rst (hunk 2 above)
Comment (andy-maier) [reply docs/usage.rst:656]: DONE

File: docs/usage.rst (hunk 3 above)
Comment (andy-maier) [reply docs/usage.rst:660]: DONE - Made the term consistent with how we define them in section "Available metrics":
* HMC metric service based
* HMC resource property based

**Pull request merged**